ktor-features/ktor-mustache/src/io/ktor/mustache/Mustache.kt

```diff
@@ -0,0 +1,90 @@
+package io.ktor.mustache
+
+import com.github.mustachejava.*
+import io.ktor.application.*
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.response.*
+import io.ktor.util.*
+import io.ktor.util.cio.*
+import kotlinx.coroutines.io.*
+
+/**
+ * Response content which could be used to respond [ApplicationCalls] like `call.respond(MustacheContent(...))
+ *
+ * @param template name of the template to be resolved by Mustache
+ * @param model which is passed into the template
+ * @param etag value for `E-Tag` header (optional)
+ * @param contentType response's content type which is set to `text/html;charset=utf-8` by default
+ */
+class MustacheContent(
+    val template: String,
+    val model: Any?,
+    val etag: String? = null,
+    val contentType: ContentType = ContentType.Text.Html.withCharset(Charsets.UTF_8)
+)
+
+/**
+ * Feature for providing Mustache templates as [MustacheContent]
+ */
+class Mustache(configuration: Configuration) {
+
+    private val mustacheFactory = configuration.mustacheFactory
+
+    class Configuration {
+        var mustacheFactory = DefaultMustacheFactory()
```

Comment:
Shouldn't it have type `MustacheFactory` instead of `DefaultMustacheFactory` so user's will be able to use other implementation classes?

Reply:
I shouldn't code after a whole day of meetings... yes that is completely right, I forgot that mustacheFactory gets the type `DefaultMustacheFactory`

Reply:
OK the configuration's `mustacheFactory` is now typed with the interface so other
Factory implementations are applicable.